In [14]:
import pandas as pd
import matplotlib.pyplot as plt   
%matplotlib inline
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
import numpy as np

In [15]:
df = pd.read_csv('salary.csv')

In [16]:
df = df.dropna(how='any',axis=0)

In [20]:
train, test = train_test_split(df, test_size=0.3)

In [24]:
X_train = train[['salary', 'exprior', 'yearsrank','market', 'degree', 'otherqual', 'position', 'male', 'Field','yearsabs' ]]
y_train = train['yearsworked']

In [25]:
model = sm.OLS(y_train,X_train).fit()

In [26]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:            yearsworked   R-squared:                       0.966
Model:                            OLS   Adj. R-squared:                  0.965
Method:                 Least Squares   F-statistic:                     976.8
Date:                Wed, 27 Mar 2019   Prob (F-statistic):          2.49e-248
Time:                        14:56:49   Log-Likelihood:                -895.70
No. Observations:                 359   AIC:                             1811.
Df Residuals:                     349   BIC:                             1850.
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
salary     -2.916e-05   2.77e-05     -1.051      0.2

### Does the model significantly predicts the dependent variable? Report the amount of variance explained (R^2) and significance value (p) to support your answer.

The model significantly predicts the dependent variable. This is shown from variance (R^2) 0.967(higher value generally indicates a better fit) with a p value of 0.106(less than 0.05 is considered to be statistically significant)


### What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?
90.6%



### What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
The coeffecient is -3.787e-05 this means that when salary increases by 1, the number of years worked increases by -3.787e-05

### What do the 95% confidence intervals [0.025, 0.975] mean?

### Calculate the expected salary for someone with 12 years’ work experience.

In [ ]:
xnew = X_train
Xnew = sm.add_constant(xnew)
ynewpred = model.predict(Xnew)


print(ynewpred)

### Calculate the expected salary for someone with 80 years’ work experience. Are there any problems with this prediction? If so, what are they?

### We have only looked at the number of years an employee has worked. What other employee characteristics might influence their salary?

### How does your model compare when running it on the test set - what is the difference in the Root Mean Square Error (RMSE) between the training and test sets?

In [ ]:
test_model = sm.OLS(y_test,X_test).fit()

In [ ]:
print_test_model = test_model.summary()
print(print_test_model)